File: Java/Gate Six/MiniParkingSystemTest.java
import org.junit.jupiter.api.Test;
//import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;
import static org.junit.jupiter.api.Assertions.assertFalse;
//import org.junit.jupiter.api.BeforeEach;

public class MiniParkingSystemTest{

	@Test
	public void testItParksAutomatically(){
		int [] parkingLot = {0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0};
		MiniParkingSystem.parkAutomatically(parkingLot);
		assertTrue(MiniParkingSystem.spaceAvailability(parkingLot, 1));
		MiniParkingSystem.parkAutomatically(parkingLot);
		assertTrue(MiniParkingSystem.spaceAvailability(parkingLot, 2));
	}

	@Test
	public void testItParksManually(){
		int [] parkingLot = {0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0};
		MiniParkingSystem.parkManually(parkingLot, 7);
		assertTrue(MiniParkingSystem.spaceAvailability(parkingLot, 7));
		MiniParkingSystem.parkManually(parkingLot, 15);
		assertTrue(MiniParkingSystem.spaceAvailability(parkingLot, 15));
	}

	@Test
	public void testItUnparks(){
		int [] parkingLot = {0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0};
		MiniParkingSystem.parkManually(parkingLot, 7);
		assertTrue(MiniParkingSystem.spaceAvailability(parkingLot, 7));
		MiniParkingSystem.parkManually(parkingLot, 15);
		assertTrue(MiniParkingSystem.spaceAvailability(parkingLot, 15));
		MiniParkingSystem.parkAutomatically(parkingLot);
		assertTrue(MiniParkingSystem.spaceAvailability(parkingLot, 1));
		MiniParkingSystem.unparkCar(parkingLot, 15);
		assertFalse(MiniParkingSystem.spaceAvailability(parkingLot, 15));
	}


}

File: Python/Gate Six/mini_parking_system_test.py
from unittest import TestCase
from mini_parking_system import *

class TestClass (TestCase):


	def test_it_parks_automatically(self):

		parking_lot = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
		park_automatically(parking_lot)
		park_automatically(parking_lot)
		self.assertTrue(space_availability(parking_lot, 1))
		self.assertTrue(space_availability(parking_lot, 2))

	def test_it_parks_manually(self):

		parking_lot = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]

		park_automatically(parking_lot)
		park_manually(parking_lot, 7)
		self.assertTrue(space_availability(parking_lot, 1))
		self.assertFalse(space_availability(parking_lot, 2))
		self.assertTrue(space_availability(parking_lot, 7))
		unpark_car(parking_lot, 1)
		self.assertFalse(space_availability(parking_lot, 1))

